### ISSUE TITLE
package configuration not applied to \ delimited paths

### ISSUE THREAD

**Issue by aluanhaddad** [jspm/jspm-cli#2371]
**JSPM Version:** 0.17.0-beta.45

**Transpiler Plugin(s):** babel

**Details:**
Package configuration is not applied when paths specified to commands such as `run` and `bundle` if `\` is used. For example this can happen on Windows if you use the autocompletion provided by `cmd.exe`.

**Config:**
```js
SystemJS.config({
  transpiler: "babel",
  baseURL: "/",
  packages: {
     "test": {
      "main": "index.ts",
      "defaultExtension": "ts",
      "format": "esm",
      "meta": {
        "*.ts": {
          "loader": "plugin-typescript"
        },
        "*.js": {
          "loader": "plugin-typescript"
        }
      }
    }
  }
});
```

Running
```shell
jspm run test/some-test.ts
```
package configuration is applied and plugin-typescript therefore is used to load the file.

However, when running
```shell
jspm run test\some-test.ts
```
plugin-typescript is not used to load the module and this results in a syntax error being raised by Babel.
Here is code itself for reference. Some TypeScript syntax is accepted by Babel but this snippet fails.
```ts
import test from 'tape';

import {StartWithValueConverter as SUT} from 'app/value-converters/array-operations';

test('example', ({deepEqual, end}) => {
  let sut: SUT; // triggers error.
  sut = new SUT();
  const array = [1, 2, 3];
  const result = sut.toView(array, 0);
  deepEqual(result, [0, 1, 2, 3]);
  end();
});
```

**Comment by guybedford** [comment]
`jspm run` is applied to module names, not file paths (just like `System.import`). 

It is inconsistent that `\` isn't supported,  as `\` is usually automatically converted into `/` in URL normalization, so we should too, but strictly speaking, this can be solved by supplying a URL-like module form.

I've created a resolver issue to track this at https://github.com/ModuleLoader/es-module-loader/issues/537.

**Comment by guybedford** [comment]
I've added a fix for this that will only work for `./test\\some-test.ts`.

There is another complexity here and that is that paths beginning with `.\\` will not be supported for relative resolution. I've posted a spec issue for this at https://github.com/whatwg/html/issues/3106.

The final case is then just supporting plain names being `\\` separated (`pkgname\\x`) instead of `pkgname/x`, which would be for integration into SystemJS itself...
